drivers/console/display_console.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2019 Intel Corp.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_PRINTK) || defined(CONFIG_STDOUT_CONSOLE)
+
+#include <device.h>
+#include <display.h>
+#include <init.h>
+#include <string.h>
+
+static struct device *display;		/* display device */
+static int size_x, size_y;		/* dimensions */
+static int current_x, current_y;	/* cursor position */
+
+/*
+ * For now, we only support the Spleen 8x16 font in lib/fonts/spleen_8x16.c.
+ * It's simple enough to make this more flexible if/when the need arises.
+ */
+
+#define FONT_WIDTH	8
+#define FONT_HEIGHT	16
+#define FONT_PITCH	8
+#define FONT_SIZE	16
+#define FONT_FIRST	32
+#define FONT_LAST	126
+
+extern u8_t font_spleen_8x16[];
+
+const static struct display_buffer_descriptor char_desc = {
+	.width = FONT_WIDTH,
+	.height = FONT_HEIGHT,
+	.pitch = FONT_PITCH,
+	.buf_size = FONT_SIZE
+};
+
+/* Why aren't these hooks declared in a header somewhere? */
```

Comment:
gross historical cruft, from back when this OS was called something else.
If we don't have any open GH enhancements to refactor how we do console setup, we really should. I think @pfalcon was going to take a stab at it at one point?

Reply:
> console setup, we really should. I think @pfalcon was going to take a stab at it at one point?

Well, I definitely (still) would like to, but it's kind of self-assigned background priority to me. Besides, good console setup would involve POSIX subsys, and it has a bunch of issues of its own (and where I'm currently at).